### PR TITLE
Guard %ENV access with exists() checks and some minor edge case bugs I found while doing so

### PR DESCRIPTION
**Pull request by demerphq**
In  https://github.com/Perl/perl5/pull/20928 I discovered that if %ENV is locked while testing and a test fails then Test::More blows up horribly.  You can argue this issue several ways, including that it is a Hash::Util testing bug. But I think Test::More can harden itself pretty reasonably against this issue by simply checking for key existence before reading a key.  I did not attempt to deal with the places where there are *writes* to ENV, obviously that is going to end badly if %ENV is locked. But reading from a locked %ENV during testing is easy to guard against and obviously at least possible to happen since code doing so has been in perl core for a long time. :-) 

Related tickets:
https://github.com/Test-More/test-more/issues/907
https://github.com/Perl/perl5/issues/20929

Along the way I noticed a subtle error in the logic for parsing the T2_FORMATTER env var. If the string is set to "+" it would end up setting the formatter to the empty string. I also realized it was possible for the logic to not set it at all, and step past the defaulting logic in Test2::API::Instance. So I fixed it. Since part of this logic reads from the %ENV I dealt with both issues in the same commit. Hope that is ok.

### CODE REVIEW

File: t/Legacy/More.t [hunk 1]
```diff
@@ -196,6 +196,10 @@ is_deeply( { foo => $sub, bar => [1, $glob] },
 is_deeply( qr/a/, qr/a/, "same regex" );
 
 
-# These two tests must remain at the end.
-is( $@, $Err,               '$@ untouched' );
-cmp_ok( $!, '==', $Errno,   '$! untouched' );
+SKIP: {
```
Comment (haarg) [inline t/Legacy/More.t:199]: I think it would be better to fix this issue. It should just require localizing these variables around the call to `warnings::enabled` that was recently added.

Comment (demerphq) [reply t/Legacy/More.t]: Done.

**Comment by demerphq** [comment]
Closing as per https://github.com/Test-More/test-more/issues/907#issuecomment-1468498621
